include/booleval/tree/tree_node.hpp

```diff
@@ -45,29 +45,21 @@ namespace tree {
  * as well as the token that the node represents in the actual expression tree.
  */
 struct tree_node {
-    token::token token;
+    token::token token{ token::token_type::unknown };
     std::shared_ptr<tree_node> left;
     std::shared_ptr<tree_node> right;
```

Comment:
Missing initialization of smart pointer to `nullptr`

Reply:
@m-peko I was under the impression that a shared_ptr will already be nullptr by default and doesn't need initialising?